packages/loopring_v3/contracts/amm/libamm/AmmData.sol

```diff
@@ -105,25 +123,21 @@ library AmmData
         uint8       feeBips;
         Token[]     tokens;
 
-        // A map of approved transaction hashes to the timestamp it was created
-        mapping (bytes32 => uint) approvedTx;
+        uint        poolTokenToBurn;
+
+        mapping (address => uint) exitLockIdx;
+        TokenLock[] exitLocks;
+        uint         exitLocksIndex;
 
-        // A map from an owner to a token to the balance
-        mapping (address => mapping (address => uint)) userBalance;
+        mapping (address => TokenLock[]) joinLocks;
+        mapping (address => uint) joinLockIdx;
 
-        // A map from an owner to the timestamp from which all funds of the user funds will be locked
-        mapping (address => uint) lockedSince;
-        // A map from an owner to the timestamp until all funds of the user are locked
-        // A zero value == locked indefinitely.
-        mapping (address => uint) lockedUntil;
+        mapping (address => mapping (address => uint96)) balance;
 
-        // A map from a token to the total balance owned directly by LPs (so NOT owned by the pool itself)
-        mapping (address => uint) totalUserBalance;
```

Comment:
Will have to come back unless user balances and pool balances (which when shutdown are force withdrawn to the owner, so also the pool contract) are kept separately.